homeassistant/components/plugwise/__init__.py

```diff
@@ -1 +1,169 @@
-"""Plugwise Climate (current only Anna) component for Home Assistant."""
+"""Plugwise platform for Home Assistant Core."""
+
+import asyncio
+from datetime import timedelta
+import logging
+from typing import Optional
+
+from Plugwise_Smile.Smile import Smile
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import DOMAIN
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSOR_PLATFORMS = ["sensor"]
+ALL_PLATFORMS = ["binary_sensor", "climate", "sensor", "switch"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Plugwise platform."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Plugwise Smiles from a config entry."""
+    websession = async_get_clientsession(hass, verify_ssl=False)
+    api = Smile(
+        host=entry.data["host"], password=entry.data["password"], websession=websession
+    )
+
+    try:
+        connected = await api.connect()
+
+        if not connected:
+            _LOGGER.error("Unable to connect to Smile")
+            raise PlatformNotReady
+
+    except Smile.PlugwiseError:
+        _LOGGER.error("Error while communicating to device")
+        raise PlatformNotReady
+
+    except asyncio.TimeoutError:
+        _LOGGER.error("Timeout while connecting to Smile")
+        raise PlatformNotReady
+
+    if api.smile_type == "power":
+        update_interval = timedelta(seconds=10)
+    else:
+        update_interval = timedelta(seconds=60)
+
+    api.get_all_devices()
```

Comment:
Is this doing blocking I/O?

Reply:
All functions handling I/O are asynced, these are just lookups in cached dictionary data.